### PR TITLE
BaSP-HOTFIX-MR-107: create consitency between all validations and fixed bugs

### PR DESCRIPTION
**Pull request by fernandofernandezuy**
This pull request addresses several modifications and bug fixes in the server. The following aspects have been modified:

• Fixed bugs that prevented the update in admin, trainers, and activities

• Created consistency between all joi validations following QA's requirements doc.

• Added functionality in the controllers to prevent sending the same data in the edit of all entities

• Created consistency between response messages

• Correction in the model of Trainer, is_active for isActive

• Create consistency between controllers, change all to async/await for better readability

### CODE REVIEW

**Review by AleMac17 — APPROVED** [review]

File: src/controllers/class.js [hunk 1]
```diff
@@ -4,60 +4,61 @@ const Class = require('../models/class');
...
+      const bodyObj = req.body;
+      const isEqual = Object.entries(bodyObj).every(([key]) => {
+        if (key !== '_id' && key !== '__v') {
+          return bodyObj[key] === classObj[key];
+        }
+        return true;
+      });
+
+      if (isEqual) {
         return res.status(404).json({
```
Comment (AleMac17) [inline src/controllers/class.js:38]: Shouldn't this be a 400 error? since 404 is for not fount cases

File: src/controllers/super-admins.js [hunk 2]
```diff
@@ -108,7 +108,7 @@ const updateSuperAdmin = (req, res) => {
         return true;
       });
       if (isEqual) {
-        return applyResponse(res, 404, 'Update rejected. ReqBody is identical for that id instance', undefined, true);
+        return applyResponse(res, 404, 'There is nothing to change', undefined, true);
```
Comment (AleMac17) [inline src/controllers/super-admins.js:111]: shouldn't this be a 400 error? since 404 is for not found cases

**Review by AleMac17 — CHANGES_REQUESTED** [review]

**Comment by fernandofernandezuy** [comment]
[[HOTFIX] - Server Validations](https://trello.com/c/0avxF8Os/107-hotfix-server-validations)

**Review by lorenzomauro59 — APPROVED** [review]
NICE job Fer!

**Review by AleMac17 — APPROVED** [review]

**Pull request merged**